### ISSUE TITLE
Enable radius filter to Delaunay spatial_neighbors

### ISSUE THREAD

**Issue by marcovarrone** [theislab/squidpy#387]
### Type of the feature

- [ ] New function in `sqduipy.im`?
- [ ] New function in `squidpy.gr`?
- [ ] New function in `squidpy.pl`?
- [x] Change an existing functionality, such as default behavior?
- [ ] Other?

### Description
The function `squidpy.gr.spatial_neighbors` already receives `radius` as a parameter for the k-nearest neighbor version, so that nodes farther than the radius value are not linked.

However, if `delaunay=True`, the `radius` parameter is ignored.
I was thinking that after computing the adjacency matrix using Delaunay triangulation we could remove the connections longer than `radius`, if a value for it is set.

**Comment by michalk8** [comment]
Any thoughts on this @giovp ? Personally, I don't think this is super intuitive + not sure about use cases, not mention there can be further confusion once the cylindrical shell in #388 is implemented.

**Comment by giovp** [comment]
I think it makes sense cause it's not intended behaviour, although through that it could be useful to have that type of filter. Considering #388 , maybe it makes sense to change the API , cause as of now arguemnts interact in a non-trivial way. Maybe we could have 

```python
sq.gr.generic_neigh()
sq.gr.grid_neigh()
```

?

**Comment by michalk8** [comment]
I'd prefer not to split the API, but clean a bit the arguments. My suggestion is:
1. merge `neigh_grid` and `n_neigh` to 1 common argument: `n_neighs: int = 6`, since they are similar enough
2. merge `radius: float` and `n_rings: int` into `r: Union[int, float, Tuple[float, float]]` - this corresponds to `min/max` radius (if a `tuple` for KNN/Delaunay graph and number of rings for grids (think conceptually are still pretty similar, though not as in 1.)

Also, would be nice to refactor [this](https://github.com/theislab/squidpy/blob/master/squidpy/gr/_build.py#L98-L127) to a separate function.

@giovp @AnnaChristina any opinions?

**Comment by MxMstrmn** [comment]
I would be cautious here as the Delaunay triangulation is a fixed mathematical concept. Removing edges that exceed a certain distance destroys the triangulation and also the construction of its dual graph (Voronoi tesselation).  If possible, I would rather try to fix API with its interacting arguments (`radius` vs. `delaunay`).

**Comment by marcovarrone** [comment]
@michalk8 my use case is that I am using the Delaunay triangulation to build the network because of some properties that are better than using KNN (e.g. two edges cannot cross each other and I don't have a fixed degree for each node). 
However, the triangulation can lead to very long edges which are meaningless in my case, so I wanted to remove them based on the distance between the two nodes.
For my type of analysis, I am not interested in the triangulation or the Voronoi tessellation. @MxMstrmn has a good point, but I thought that by making the `radius` parameter optional one could still obtain the "classical" Delaunay triangulation.

In my opinion, if Delaunay has no radius I would rather split the method into:
```
sq.gr.radius_neighs()
sq.gr.delaunay_neighs()
```
because, if I understood well how `sq.gr.spatial_neigbhbors()` works, there are not many parameters in common between KNN and Delaunay, while there is a lot of similarity between generic and grid behaviors, as @michalk8 pointed out.

**Comment by michalk8** [comment]
closed via #393
